src/genie/libs/parser/iosxr/show_controllers.py

```diff
@@ -1286,3 +1287,481 @@ def cli(self, unit=0, location='all', output=None):
+
+    def cli(self, ethernet_interface_type=None, interface_path_id=None, output=None):
+        if output is None:
+            if ethernet_interface_type and ethernet_interface_type.lower() in self.valid_ethernet_interface_types:
+                if interface_path_id:
+                    cmd = self.cli_command[0].format(ethernet_interface_type=ethernet_interface_type, 
+                    interface_path_id=interface_path_id)
+                    out = self.device.execute(cmd)
+                    return self.cli_parse(out, ethernet_interface_type, interface_path_id)
+                else:
+                    interfaces_list = self.cli_get_interfaces(ethernet_interface_type)
+            else:
+                interfaces_list = []
+                for ethernet_interface_type in self.valid_ethernet_interface_types:
+                    interfaces_list += self.cli_get_interfaces(ethernet_interface_type)
+            return_dict = {}
+            for ethernet_interface_type, interface_path_id in interfaces_list:
+                cmd = self.cli_command[0].format(ethernet_interface_type=ethernet_interface_type, 
+                    interface_path_id=interface_path_id)
+                out = self.device.execute(cmd, error_pattern=[])
+                return_dict.update(self.cli_parse(out, ethernet_interface_type, interface_path_id))
+            return return_dict
+        else:
+            out = output
+            if ethernet_interface_type and ethernet_interface_type.lower() in self.valid_ethernet_interface_types:
+                if interface_path_id:
+                    return self.cli_parse(out, ethernet_interface_type, interface_path_id)
+                else:
+                    interfaces_list = self.cli_get_interfaces(ethernet_interface_type)
+            else:
+                interfaces_list = []
+                for ethernet_interface_type in self.valid_ethernet_interface_types:
+                    interfaces_list += self.cli_get_interfaces(ethernet_interface_type)
+            return_dict = {}
+            for ethernet_interface_type, interface_path_id in interfaces_list:
+                return_dict.update(self.cli_parse(out, ethernet_interface_type, interface_path_id))
+            return return_dict
+
+
+    def cli_get_interfaces(self, ethernet_interface_type, output=None):
+        "get interfaces from ethernet_interface_type"
+        cmd = f"show controllers {ethernet_interface_type} ?"
+        if output is None:
+            out = self.device.execute(cmd, error_pattern=[])
+        else:
+            out = output
+        p1 = re.compile(r'(?P<interface_path_id>\d+\/\d+\/\d+\/\d+)\s+\S+\s+Interface Instance')
+        interface_list = []
+        for line in out.splitlines():
+            line = line.strip()
+            m = p1.match(line)
+            if m:
+                group = m.groupdict()
+                interface_list.append((ethernet_interface_type, group['interface_path_id']))
+        return interface_list
```

Comment:
We typically try to avoid having parsers executing other commands

Please try to find a way to parse the output without secondary commands. If impossible, then we can discuss